### PR TITLE
Pass `-amd` (also make dependents) when building native-images

### PR DESCRIPTION
**Pull request by zakkak**
This allows multi-module directories like `test-extension` to be tested as expected.

Related to discussion in https://github.com/quarkusio/quarkus/pull/30027#issuecomment-1362723823

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Good idea!

**Review by gsmet — CHANGES_REQUESTED** [review]
Won't it rebuild a large part of the tree every time? The idea was to avoid that.

**Comment by geoand** [comment]
> Won't it rebuild a large part of the tree every time

Why would it? This should only build the dependents (not dependencies) of the root test module. At least that's how I understand it and what I saw when running it.

**Review by gsmet — APPROVED** [review]
Ah sorry, it's `-amd`, not `-am`. /me go hiding.

**Comment by famod** [comment]
I'm currently running some pre-christmas errands. Will have a look this evening, in case you want to wait.

**Review by famod — APPROVED** [review]
Given that `-pl` is not recursive in Maven 3, meaning it won't execute submodules (as opposed to `-f`), `-amd` can indeed be used to work around that (as long as submodules have a `parent` declaration, hence depend on the parent).
(Btw, `pl` does also take paths, not only the artifact id (referring to https://github.com/quarkusio/quarkus/pull/30027#issuecomment-1362788828))

Now, the question is: when are we expecting `test-extension` to be selected? Thing is that a changed `core-deployment` (as in #30027) will impact `test-extension/extension/runtime`, `test-extension/extension/deployment` and `test-extension/tests`, but not their parent `test-extension` (or `test-extension/extension`). Looks like I have to add some "parent matching" logic to `filter-native-tests-json.sh` to mitigate that problem.
Additionally, in case of #30027 it's not `filter-native-tests-json.sh` that is swallowing `test-extension` because I can see in CI logs that the input for this script doesn't even include `test-extension/*` in the first place.
Somehow (in the initial JDK 11 build), GIB is not detecting any of those `test-extension` modules as being impacted.
I'll have a look at that too.

Long story short: This PR does make sense, but `-amd` will not make any difference for many cases as long as the above issues are still present.

**Comment by famod** [comment]
A follow-up to https://github.com/quarkusio/quarkus/pull/30035#pullrequestreview-1228411257:

> Additionally, in case of #30027 it's not `filter-native-tests-json.sh` that is swallowing `test-extension` because I can see in CI logs that the input for this script doesn't even include `test-extension/*` in the first place. Somehow (in the initial JDK 11 build), GIB is not detecting any of those `test-extension` modules as being impacted. I'll have a look at that too.

Turns out everything is fine here, those `test-extension/*` _are_ detected as impacted, I just didn't search properly (don't use the browser search function!).

> Looks like I have to add some "parent matching" logic to `filter-native-tests-json.sh` to mitigate that problem.

Yes and no. The problem is indeed a mismatch between script input and json content but the solution is the other way around.
I'll test my fix in my fork, expect a PR soon.
